packages/contracts/contracts/ETHVirtualAppAgreement.sol

```diff
@@ -40,50 +33,31 @@ contract ETHVirtualAppAgreement {
       "agreement lockup time has not elapsed"
     );
 
-    Transfer.Transaction memory resolution = agreement.registry
+    bytes memory resolution = agreement.registry
       .getResolution(agreement.appIdentityHash);
 
-    require(
-      agreement.terms.assetType == resolution.assetType,
-      "returned incompatible resolution based on bad asset type"
-    );
-
-    require(
-      agreement.terms.token == resolution.token,
-      "returned incompatible resolution based on bad token value"
-    );
-
-    require(
-      agreement.capitalProvided > resolution.value[0],
-      "returned incompatible resolution"
-    );
+    uint256 resolutionAsValue = abi.decode(resolution, (uint256));
```

Comment:
Does `uint256(resolution)` do the same thing as this line?

Reply:
Not sure - used `abi.decode` for consistency